### DISCUSSION TITLE
StageBackEdge and StageFrontEdge triggers

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1151] [Feature Request]
Currently the usage of these triggers is not very intuitive. If you walk a character to the very end of a stage, the stage edge triggers will not return 0 but instead the value of the stage's ScreenLeft or ScreenRight.

I think these triggers should be adjusted so that they return 0 when a character is at the ends of the stage. This is both more intuitive and consistent with the way Back and FrontEdge triggers work.

The other benefit of this is that when BackEdgeDist = StageBackEdge, it would mean that the camera can't scroll further in that direction.

Also (and nitpicking starts here) maybe the actual trigger names should include "dist" in them as well.

**Comment by Lazin3ss** [comment]
I thought these triggers already worked like this 🤔 it seems it isn't like that. I'm in favor of this change for 0.99

**Comment by potsmugen** [comment]
Implemented.
